tests/test_dataset.py

```diff
@@ -51,14 +51,35 @@ def test_shape(self, expected_shape):
         dataset = Dataset(data=test_data, transform=LoadNiftid(keys=['image', 'label', 'extra']))
         data1 = dataset[0]
         data2 = dataset[1]
+        dataset_precached = Dataset(data=test_data, transform=LoadNiftid(keys=['image', 'label', 'extra']), cache_dir=tempdir)
```

Comment:
If you want to test hash computation, suggest testing with **float** data instead of `randint`.

Reply:
I am not hashing the tensor values.  I agree with your statement above that it would be problematic.  Anyway, if we need to load the data and apply the transforms in order to compute the hash, that nullifies the benefits of persistence layer completely.

Reply:
I think we need to test whether it still works fine if the user input data buffer directly instead of file names.

Reply:
It probably won't.  This solution was predicated on the assumption that inputs would be file names.  It won't work for buffered data.  Perhaps the use of the cache_dir can only be used in the very common case of filenames, and dis-allowed for any non-filestring inputs.

Reply:
I propose that we simply disable the persistant behavior for data buffers.